jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/caws/DevEnvStatusWatcher.kt

```diff
@@ -0,0 +1,120 @@
+            val initialEnv = client.getDevEnvironment {
+                it.id(envId)
+                it.spaceName(org)
+                it.projectName(projectName)
+            }
+            val inactivityTimeout = initialEnv.inactivityTimeoutMinutes()
+            if (inactivityTimeout == 0) {
+                LOG.info("Dev environment inactivity timeout is 0, not monitoring")
+                return@launch
+            }
+            val inactivityTimeoutInSeconds = inactivityTimeout * 60
+
+            // ensure the JetBrains inactivity tracker and the activity api are in sync
+            val jbActivityStatusJson = UnattendedStatusUtil.getStatus()
+            val jbActivityStatus = jbActivityStatusJson.projects?.first()?.secondsSinceLastControllerActivity ?: 0
+            notifyBackendOfActivity((getActivityTime(jbActivityStatus).toString()))
+            var secondsSinceLastControllerActivity = jbActivityStatus
+
+            while (true) {
+                val statusJson = UnattendedStatusUtil.getStatus()
+                val lastActivityTime = statusJson.projects?.first()?.secondsSinceLastControllerActivity ?: 0
+
+                if (lastActivityTime < secondsSinceLastControllerActivity) {
+                    // update the API in case of any activity
+                    notifyBackendOfActivity((getActivityTime(lastActivityTime).toString()))
+                }
+                secondsSinceLastControllerActivity = lastActivityTime
+
+                val lastRecordedActivityTime = CawsEnvironmentClient.getInstance().getActivity()?.timestamp
+                if(lastRecordedActivityTime == null) {
+                    LOG.error("Couldn't retrieve last recorded activity from API")
+                    return@launch
+                }
+                val durationRecordedSinceLastActivity = Instant.now().toEpochMilli().minus(lastRecordedActivityTime.toLong())
+                val secondsRecordedSinceLastActivity = durationRecordedSinceLastActivity / 1000
+
+                if (secondsRecordedSinceLastActivity >= (inactivityTimeoutInSeconds - 300)) {
+                    try {
+                        val inactivityDurationInMinutes = secondsRecordedSinceLastActivity / 60
+                        val ans = runBlocking {
+                            val continueWorking = withContext(getCoroutineUiContext()) {
+                                return@withContext MessageDialogBuilder.okCancel(
+                                    message("caws.devenv.continue.working.after.timeout.title"),
+                                    message("caws.devenv.continue.working.after.timeout", inactivityDurationInMinutes)
+                                ).ask(project)
+                            }
+                            return@runBlocking continueWorking
+                        }
+
+                        if (ans) {
+                            notifyBackendOfActivity(getActivityTime(0).toString())
```

Comment:
nit: not obvious what the `0` means. maybe make 0 the default value and rename the parameter name to "secondsInThePast" or sometihng

Reply:
Done, changed the implementation to have zero as the default value along with a change in the variable name